src/GraphQL/Validation/Rules.Custom/InputFieldsAndArgumentsOfCorrectLength.cs

```diff
@@ -0,0 +1,81 @@
+using System.Threading.Tasks;
+using GraphQL.Execution;
+using GraphQL.Language.AST;
+using GraphQL.Types;
+using GraphQL.Validation.Errors;
+
+namespace GraphQL.Validation.Rules
+{
+    /// <summary>
+    /// Validation rule that checks minimum and maximum length of provided values for input fields and
+    /// arguments that marked with <see cref="LengthDirective"/> directive. Doesn't check default values.
+    ///
+    /// <br/><br/>
+    /// This is not a standard validation rule that is not in the specification.
+    /// </summary>
+    public class InputFieldsAndArgumentsOfCorrectLength : IValidationRule
```

Comment:
This is great demo code, and we should include it as a test to demonstrate a fully-functional directive, and it can serve to help us realize where the framework should change to suit these types of use cases .... but I don't think it should be included in the library.  Especially not with what appears to be some major design flaws as previously noted.

Reply:
>  design flaws

It's not that bad. Existing code will not be harmed at all. No performance loss.

> I don't think it should be included in the library

Why so? Think of many scalar types that we have included in the library - `Uri` for example. I do not mind moving this directive into a separate project, but is it necessary now? There are other parts of the core library that could be moved to other projects to make a more modular solution. This is a lot of work and you can't do it quickly.

Reply:
Design flaws were mitigated.